src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -362,95 +364,117 @@ MultiPlanTree(Query *queryTree)
 static void
 ErrorIfQueryNotSupported(Query *queryTree)
 {
-	char *errorDetail = NULL;
+	char *errorMessage = NULL;
 	bool hasTablesample = false;
 	bool hasUnsupportedJoin = false;
 	bool hasComplexJoinOrder = false;
 	bool hasComplexRangeTableType = false;
 	bool preconditionsSatisfied = true;
+	const char *errorDetail = NULL;
+	const char *joinDetail = "Consider joining tables on partition column and have "
+							 "equal filter on joining columns.";
+	const char *filterDetail = "Consider using an equality filter on the distributed "
+							   "table's partition column.";
 
 	if (queryTree->hasSubLinks)
 	{
 		preconditionsSatisfied = false;
-		errorDetail = "Subqueries other than in from-clause are currently unsupported";
+		errorMessage = "cannot run distributed queries with subqueries other than "
+					   "in from-clause";
+		errorDetail = filterDetail;
 	}
 
 	if (queryTree->hasWindowFuncs)
 	{
 		preconditionsSatisfied = false;
-		errorDetail = "Window functions are currently unsupported";
+		errorMessage = "cannot run distributed queries with window functions";
+		errorDetail = filterDetail;
 	}
 
 	if (queryTree->setOperations)
 	{
 		preconditionsSatisfied = false;
-		errorDetail = "Union, Intersect, or Except are currently unsupported";
+		errorMessage = "cannot run distributed queries with union, intersect, or except";
+		errorDetail = filterDetail;
 	}
 
 	if (queryTree->hasRecursive)
 	{
 		preconditionsSatisfied = false;
-		errorDetail = "Recursive queries are currently unsupported";
+		errorMessage = "cannot run distributed queries with recursive";
```

Comment:
_recursive_ is an adjective. _queries with recursive_ is pretty weird. Uppercase all keywords.